beps/con-jms/1365_support_jms_destination_per_svc.md

```diff
@@ -0,0 +1,153 @@
+
+* Increased configuration complexity.
+* Redundant listener instances.
+* Reduced runtime efficiency due to unnecessary connection creation.
+
+By enabling **queue/topic per service** rather than per listener, we allow the developer to configure the queue/topic-specific behavior at the service level while keeping the connection setup at the listener level.
+
+## Description
+
+### Current Model:
+
+```ballerina
+listener jms:Listener jmsListener = check new ({
+    connectionConfig: {
+        initialContextFactory: "org.apache.activemq.jndi.ActiveMQInitialContextFactory", 
+        providerUrl: "tcp://localhost:61616",
+        connectionFactoryName: "ConnectionFactory"
+    },
+    acknowledgementMode: jms:AUTO_ACKNOWLEDGE,
+    consumerOptions: {
+        destination: {
+            'type: jms:TOPIC,
+            name: "test-topic"
+        }
+        ...
+    }
+});
+
+service "consumer-service" on jmsListener {
+    remote function onMessage(jms:Message message) { ... }
+}
+```
+
+### Proposed Model:
+
+We propose to **decouple the destination, session, and consumer configurations from the listener**, and move them into **service-level annotations**, enabling **a single listener to handle multiple JMS services**.
+
+```ballerina
+listener jms:Listener jmsListener = check new ({
+  initialContextFactory: "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+  providerUrl: "tcp://localhost:61616",
+  connectionFactoryName: "ConnectionFactory"
+});
+
+@jms:ServiceConfig {
+  acknowledgementMode: jms:AUTO_ACKNOWLEDGE,
+  config: {
+    topicName: "topic-1"
+    ...
+  }
+}
+service "topic-1-consumer" on jmsListener {
+  remote function onMessage(jms:Message message) { ... }
+}
+
+@jms:ServiceConfig {
+  acknowledgementMode: jms:AUTO_ACKNOWLEDGE,
+  config: {
+    topicName: "topic-2"
+    ...
+  }
+}
+service "topic-2-consumer" on jmsListener {
+  remote function onMessage(jms:Message message) { ... }
+}
+```
+
+* The destination details—such as the destination type (queue or topic) and the destination name (the actual name of the queue or topic)—are specified within the `jms:ServiceConfig` annotation. In this approach, the **service name does not need to match the destination name**.
+
+### Key Changes
+
+* To initialize the `jms:Listener`, the user should provide only the `jms:ConnectionConfiguration`. Hence, the listener `init` function will be updated as follows:
+
+  ```ballerina
+  public isolated function init(*jms:ConnectionConfiguration connectionConfig) returns Error?;
+  ```
+
+* Introduce a new service-level annotation `jms:ServiceConfig`.
+
+
+  ```ballerina
+  # Represents configurations for a JMS queue.
+  #
+  # + queueName - The name of the queue to consume messages from
+  # + messageSelector - Only messages with properties matching the message selector expression are delivered. 
+  #                     If this value is not set that indicates that there is no message selector for the message consumer
```

Comment:
Shall we add an example of messageSelector here? which will help BI users as well